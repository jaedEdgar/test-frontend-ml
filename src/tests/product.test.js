import React from 'react';
import renderer from 'react-test-renderer';
import Product from "../client/components/product"
import '@testing-library/jest-dom'
import { MemoryRouter as Router, withRouter } from 'react-router-dom'

const mock = {
  amount: "80999",
  city_name: "Rosario",
  decimals: 0,
  free_shipping: true,
  id: "MLA909240269",
  picture: "http://http2.mlstatic.com/D_998849-MLA44698052458_012021-I.jpg",
  title: "Xiaomi Poco X3 Dual Sim 64 Gb Out Of The Blue 6 Gb Ram",
}


it('will pass', () => {
  expect(
    renderer.create(<Router>
      <Product
        picture={mock.picture}
        title={mock.title}
        amount={mock.amount}
        decimals={mock.decimals}
        id={mock.id}
        city_name={mock.city_name}
        free_shipping={mock.free_shipping} />
    </Router>)
      .toJSON()
  ).toMatchSnapshot()
})