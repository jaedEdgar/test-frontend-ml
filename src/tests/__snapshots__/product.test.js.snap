// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will pass 1`] = `
<li
  className="result-search--item"
>
  <div
    className="result-search--preview"
  >
    <img
      alt="Xiaomi Poco X3 Dual Sim 64 Gb Out Of The Blue 6 Gb Ram"
      src="http://http2.mlstatic.com/D_998849-MLA44698052458_012021-I.jpg"
    />
  </div>
  <div
    className="result-search--wrapp"
  >
    <div
      className="result-search--description"
    >
      <div
        className="result-search--description-price"
      >
        <span
          className="result-search--description-mount"
        >
          $ 
          80,999
        </span>
        <i
          className="free-shipping-icon"
          title="EnvÃ­o gratis"
        />
      </div>
      <h2>
        <a
          className="result-search--description-title"
          href="/items/MLA909240269"
          onClick={[Function]}
        >
          Xiaomi Poco X3 Dual Sim 64 Gb Out Of The Blue 6 Gb Ram
        </a>
      </h2>
    </div>
    <span
      className="result-search--description-city"
    >
      Rosario
    </span>
  </div>
</li>
`;
